src/plugins/intel_cpu/src/nodes/gather.cpp

```diff
@@ -41,8 +42,64 @@ bool Gather::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std
     return true;
 }
 
+class GatherShapeInfer : public ShapeInferEmptyPads {
+public:
+    GatherShapeInfer(bool isAxisInputConst, int axis, int batchDims, IShapeInfer::port_mask_t port_mask) :
+                     m_isAxisInputConst(isAxisInputConst), m_axis(axis), m_batchDims(batchDims), m_port_mask(port_mask) {}
+    Result infer(const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
+                 const std::unordered_map<size_t, MemoryPtr>& data_dependency) override {
+        static constexpr size_t GATHER_DATA = 0, GATHER_INDICES = 1, GATHER_AXIS = 2;
+
+        auto input_shape = input_shapes[GATHER_DATA].get();
+        auto indices_shape = input_shapes[GATHER_INDICES].get();
+
+        if (!m_isAxisInputConst)
+            m_axis = reinterpret_cast<const int32_t *>(data_dependency.at(GATHER_AXIS)->GetPtr())[0];
```

Comment:
I think it won't be superfluous to check the memory precision, to make sure it is I32 and throw if it isn't.

Reply:
Applied. Thanks Maksim!